### PR TITLE
Remove obsolete v1 tests (part 1)

### PR DESCRIPTION
**Pull request by jason-fox**
In order to merge https://github.com/telefonicaid/iotagent-node-lib/pull/995 I want to get the unit-tests running without relying on NGSI-v1.  This PR just deletes tests which are already duplicated in the NGSI-v2 test suite. Any drop in coverage should be within acceptable limits - this one is 0.2%

The failure is due to #550 . The coverage can be seen here: https://coveralls.io/github/jason-fox/iotagent-json?branch=master

### CODE REVIEW

**Comment by jason-fox** [comment]
Part 2 will run the remaining v1 tests as NGSI v2.

**Comment by fgalan** [comment]
> Part 2, which will be **a separate PR** will switch the remaining v1 tests to run as NGSI v2 and change expectations accordingly. It is just easier to define necessary changes without still having unnecessary failing dead v1 tests in the codebase.

In other similar PRs (https://github.com/telefonicaid/lightweightm2m-iotagent/pull/248 and https://github.com/telefonicaid/iotagent-ul/pull/478), before seens this comment :), I have suggested to use `jason-fox:feature/ngsi-v1` as dependency and have all them in the same PR. I understand that your approach is to do that but for the "part 2" PR. Is my understanding correct?

However, please sync this PR with master. Some test has been fixed recently in master branch (PR #476 )

**Comment by jason-fox** [comment]
I can sync this, but the PR should be a two parter - this first step is just deleting the duplicates. The idea being that coverage doesn't drop significantly from this part.

**Comment by fgalan** [comment]
> Any drop in coverage should be within acceptable limits - this one is 0.2%

According to reports in GitHub Actions, coverage would drop 2.4%. It seems to be a bit high...

**Comment by jason-fox** [comment]
> According to reports in GitHub Actions, coverage would drop 2.4%. It seems to be a bit high...

This indicates that one of the deleted tests hasn't been duplicated by v2 ... yet :smile:

Based on the coverage change I should be able to work out which one that is.

**Comment by jason-fox** [comment]
@fgalan - this one is now only down 0.4% - I think that is done.

**Comment by fgalan** [comment]
> 
> 
> > According to reports in GitHub Actions, coverage would drop 2.4%. It seems to be a bit high...
> 
> This indicates that one of the deleted tests hasn't been duplicated by v2 ... yet 😄
> 
> Based on the coverage change I should be able to work out which one that is.

Great! The coverage metric would help to check this kind of "gaps" in NGSIv2 test suite :)

At the present moment the decrease is 0.4%. Slight above the 0.2%, but fine with me. Do you think the PR is ready to be merged or are you stilling looking for "gaps" that could explain that 0.4% drop?

**Comment by fgalan** [comment]
> At the present moment the decrease is 0.4%. Slight above the 0.2%, but fine with me. Do you think the PR is ready to be merged or are you stilling looking for "gaps" that could explain that 0.4% drop?

Ups... I haven't read your "I think that is done". Ok!

**Comment by jason-fox** [comment]
Lines like [this](https://coveralls.io/builds/38766080/source?filename=lib%2FcommonBindings.js#L98)  would seem to indicate that the remaining reduction in coverage is due to NGSI-v1 specific code (Timestamp Constants and Error handling) so I don't think I can manage better than -0.4%

File: test/unit/HTTP_reveice_measures-test.js [hunk 1]
```diff
@@ -1,339 +0,0 @@
...
-                    should.not.exist(error);
-                    const parsedBody = JSON.parse(body);
-
-                    response.statusCode.should.equal(200);
-                    should.exist(parsedBody.devices[0].transport);
-                    parsedBody.devices[0].transport.should.equal('HTTP');
-                    done();
-                });
-            });
-        });
```
Comment (mapedraza) [inline test/unit/HTTP_reveice_measures-test.js:337]: This "it" scenario is not checked on [V2 test file equivalent](https://github.com/telefonicaid/iotagent-json/blob/master/test/unit/ngsiv2/HTTP_reveice_measures-test.js), could you adapt it?

File: test/unit/HTTP_reveice_measures-test2.js [hunk 2]
```diff
@@ -1,385 +0,0 @@
...
-                    should.not.exist(error);
-                    const parsedBody = JSON.parse(body);
-
-                    response.statusCode.should.equal(200);
-                    should.exist(parsedBody.devices[0].transport);
-                    parsedBody.devices[0].transport.should.equal('HTTP');
-                    done();
-                });
-            });
-        });
```
Comment (mapedraza) [inline test/unit/HTTP_reveice_measures-test2.js:383]: This "it" scenario is not checked on [V2 test file equivalent](https://github.com/telefonicaid/iotagent-json/blob/master/test/unit/ngsiv2/HTTP_reveice_measures-test2.js), could you adapt it?

**Review by mapedraza — COMMENTED** [review]
Some diferences between the tests on diferent versions

File: test/unit/HTTP_reveice_measures-test2.js (hunk 2 above)
Comment (jason-fox) [reply test/unit/HTTP_reveice_measures-test2.js:383]: Fixed 45e83ef

File: test/unit/HTTP_reveice_measures-test.js (hunk 1 above)
Comment (jason-fox) [reply test/unit/HTTP_reveice_measures-test.js:337]: Fixed 45e83ef

**Comment by jason-fox** [comment]
So adding the additional expectations has **reduced** the coverage 🤔 

Actually it looks like the AMPQ [reconnect](https://coveralls.io/builds/38795726/source?filename=lib%2Fbindings%2FAMQPBinding.js#L162) is sometimes firing, sometimes not. I guess with more tests they likelyhood of a re-connect is higher.

**Comment by jason-fox** [comment]
Merged - the flaky AMPQ test timed out though. IThe same test passed the unit test runs, but not the coverage run - maybe give it another kick to see if it is just a timing issue.

**Comment by fgalan** [comment]
> Merged - the flaky AMPQ test timed out though. IThe same test passed the unit test runs, but not the coverage run - maybe give it another kick to see if it is just a timing issue.

At the second try it worked :)

Interestingly, coverage is no ok at GitActions (green mark) although the increase (+) or decrease (-) doesn't appear... Just curiosity I haved tried for get it from the [console log of the job](https://github.com/telefonicaid/iotagent-json/pull/553/checks?check_run_id=2392752421) but I found it.

(Not very important, anyway)

**Comment by mapedraza** [comment]
LGTM

**Review by mapedraza — APPROVED** [review]

**Pull request merged**